### ISSUE TITLE
ONNX for TensorRT — Day 24: Define bounded optimization profiles

### ISSUE THREAD

**Issue by rui-ren** [rui-ren/mobius#37]
## Objective

- [ ] Define bounded TensorRT optimization profiles that cover a Mobius model's dynamic inputs and separate low-latency robotics prefill, decode, and vision workloads.

## Brief concept

TensorRT needs a minimum, optimum, and maximum shape for every dynamic input before it can build an engine. The optimum shape guides tactic selection; minimum and maximum define the accepted runtime envelope and influence memory and build cost. Keep profiles narrow and workload-specific: batch-1 decode, prompt prefill, and camera/image processing have different useful ranges. Work on branch `ruiren/tensorRT-mobius`; this lesson is read-only and must not alter repository files.

## Steps (30–60 minutes)

- [ ] **Inventory real dynamic dimensions (10 min).** Read `dailys/day01/build_model_onnx_ir.py`, `src/mobius/tasks/_causal_lm.py`, and `src/mobius/tasks/_vision_language.py`. Record the contracts for tiny MLP `x: [batch, 4]`; causal-LM `input_ids`, `attention_mask`, `position_ids`, and every KV-cache input; and VLM `pixel_values`/`grid_thw`. Note that every dynamic input—not only `input_ids`—needs profile bounds.
- [ ] **Define three deployment envelopes (10–15 min).** Write min/opt/max values for: (a) batch-1 robot decode with `sequence_len=1` and growing `past_sequence_len`; (b) batch-1 prompt prefill with `past_sequence_len=0` and a bounded prompt range; and (c) vision input using either fixed image sizes such as `[1,3,H,W]` or the Mobius packed contract (`total_patches`, `num_images`). For every axis enforce `min <= opt <= max`, preserve fixed dimensions, and make `attention_mask` length equal `past_sequence_len + sequence_len`.
- [ ] **Explain the design choices (10 min).** For each profile, state the expected common case, why it is the optimum, the maximum latency/memory envelope, and the out-of-profile behavior. Explain why one huge profile spanning all prompt, decode, and image cases can increase tactic-search cost and produce poor robotics latency.
- [ ] **Verify the mechanism on the tiny graph (10–15 min).** Run TensorRT against the existing model without modifying it:
  ```bash
  trtexec \
    --onnx=dailys/day01/onnx/tiny_mlp_ir_v2.onnx \
    --skipInference \
    --minShapes=x:1x4 \
    --optShapes=x:8x4 \
    --maxShapes=x:32x4
  ```
  Then compare those flags with the all-or-none validation implemented in `dailys/day01/tensorrt_validation.py`.

## Deliverable

- [ ] Add one comment to this issue containing a compact profile specification for `decode`, `prefill`, and `vision`: list every input's min/opt/max shape, the intended workload, and one sentence justifying each optimum. Do not commit files.

## Verification

- [ ] Completion passes when the tiny graph builds successfully and the written profiles satisfy all of these checks: every dynamic input is covered; all fixed axes stay fixed; `min <= opt <= max` axis-by-axis; related sequence/cache dimensions are consistent; and no expected robot request lies outside its selected profile.

## Robotics relevance

Bounded workload-specific profiles make TensorRT engine memory and latency predictable enough for batch-1 perception and action loops on NVIDIA edge hardware.
